fleet/agent_group_service.go

```diff
@@ -235,6 +235,33 @@ func (svc fleetService) RemoveAgentGroup(ctx context.Context, token, groupId str
 		svc.logger.Error("removing agents from group channel", zap.Error(errors.Wrap(ErrMaintainAgentGroupChannels, err)))
 	}
 
+	group, err := svc.agentGroupRepository.RetrieveByID(ctx, groupId, ownerID)
+	if err != nil {
+		return err
+	}
+
+	limit, err := svc.agentRepo.RetrieveAllByAgentGroupID(ctx, ownerID, groupId, false)
+	if err != nil {
+		return err
+	}
```

Comment:
If you are using this just for the limit, I think we should add a query to fetch just the count, it is way faster.